core/services/keeper/upkeep_executer.go

```diff
@@ -224,7 +225,9 @@ func (ex *UpkeepExecuter) execute(upkeep UpkeepRegistration, head *evmtypes.Head
 
 	// Only after task runs where a tx was broadcast
 	if run.State == pipeline.RunStatusCompleted {
-		err := ex.orm.SetLastRunHeightForUpkeepOnJob(ex.job.ID, upkeep.UpkeepID, head.Number, pg.WithParentCtx(ctxService))
+		i, ok := KeeperList[upkeep.Registry.FromAddress.Address()]
```

Comment:
ah, I see it's also used here. We probably need to find another way to share this data.